packages/pds/tests/phone-verification.test.ts

```diff
@@ -0,0 +1,156 @@
+import assert from 'assert'
+import AtpAgent from '@atproto/api'
+import * as crypto from '@atproto/crypto'
+import { TestNetworkNoAppView } from '@atproto/dev-env'
+import { AppContext } from '../src'
+
+describe('phone verification', () => {
+  let network: TestNetworkNoAppView
+  let ctx: AppContext
+  let agent: AtpAgent
+
+  let verificationCodes: Record<string, string>
+  let sentCodes: { number: string; code: string }[]
+
+  beforeAll(async () => {
+    network = await TestNetworkNoAppView.create({
+      dbPostgresSchema: 'phone_verification',
+      pds: {
+        phoneVerificationRequired: true,
+        twilioAccountSid: 'ACXXXXXXX',
+        twilioAuthToken: 'AUTH',
+        twilioServiceSid: 'VAXXXXXXXX',
+      },
+    })
+    ctx = network.pds.ctx
+    assert(ctx.twilio)
+    verificationCodes = {}
+    sentCodes = []
+    ctx.twilio.sendCode = async (number: string) => {
+      if (!verificationCodes[number]) {
+        const code = crypto.randomStr(4, 'base10').slice(0, 6)
+        verificationCodes[number] = code
+      }
+      const code = verificationCodes[number]
+      sentCodes.push({ code, number })
+    }
+    ctx.twilio.verifyCode = async (number: string, code: string) => {
+      if (verificationCodes[number] === code) {
+        delete verificationCodes[number]
+        return true
+      }
+      return false
+    }
```

Comment:
I believe there's some comparable mocking code in dev-env—could we make use of that here?